src/protocols/gwmp/hpr_gwmp_worker.erl

```diff
@@ -0,0 +1,500 @@
+%%%-------------------------------------------------------------------
+%%% @author jonathanruttenberg
+%%% @copyright (C) 2022, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 08. Aug 2022 3:24 PM
+%%%-------------------------------------------------------------------
+-module(hpr_gwmp_worker).
+-author("jonathanruttenberg").
+
+-behaviour(gen_server).
+
+-include("../../grpc/autogen/server/packet_router_pb.hrl").
+
+-include("semtech_udp.hrl").
+
+%% API
+-export([
+    start_link/1,
+    push_data/4,
+    pubkeybin_to_mac/1
+]).
+
+%% gen_server callbacks
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2,
+    code_change/3
+]).
+
+-define(SERVER, ?MODULE).
+
+-define(METRICS_PREFIX, "helium_packet_router_").
+
+-type pull_data_map() :: #{
+    gwmp_udp_socket:socket_dest() => #{timer_ref := reference(), token := binary()}
+}.
+
+-record(state, {
+    location :: no_location | {pos_integer(), float(), float()} | undefined,
```

Comment:
We can remove this from the worker for now.

Reply:
Done.